chain/epoch-manager/src/validator_selection.rs

```diff
@@ -390,6 +338,44 @@ impl Ord for OrderedValidatorStake {
     }
 }
 
+mod old_validator_selection {
+    use super::*;
+
+    pub fn assign_chunk_producers_to_shards(
```

Comment:
❤️ Thanks for refactoring this into a new method! The `proposals_to_epoch_info` method is so huge and annoying, this is helping!